File: app/src/main/java/com/example/breezil/favnews/ui/MainActivity.java
package com.example.breezil.favnews.ui;

import android.arch.lifecycle.ViewModelProvider;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.support.v4.app.Fragment;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.example.breezil.favnews.R;
import com.example.breezil.favnews.databinding.ActivityMainBinding;
import com.example.breezil.favnews.ui.adapter.PagerAdapter;
import com.example.breezil.favnews.utils.helpers.BottomNavigationHelper;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.DaggerAppCompatActivity;
import dagger.android.support.HasSupportFragmentInjector;

public class MainActivity extends DaggerAppCompatActivity implements HasSupportFragmentInjector {


    @Inject
    ViewModelProvider.Factory viewModelFactory;
    ActivityMainBinding binding;


    @Inject
    DispatchingAndroidInjector<Fragment> dispatchingAndroidInjector;

    PagerAdapter pagerAdapter;




    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);


        setupAdapter();
        setupBottomNavigation();
        setSupportActionBar((Toolbar) binding.mainToolbar);
        getSupportActionBar().setTitle(getString(R.string.app_name));
    }

    private void setupBottomNavigation() {

        BottomNavigationHelper.disableShiftMode(binding.bottomNavViewBar);


        Menu menu = binding.bottomNavViewBar.getMenu();
        MenuItem menuItem= menu.getItem(0);
        menuItem.setChecked(true);


        /*
         * here sets the navigations to its corresponding activities
         */
        binding.bottomNavViewBar.setOnNavigationItemSelectedListener(item -> {
            switch (item.getItemId()){

                case R.id.main:

                    break;
                case R.id.search:
                    startActivity(new Intent(MainActivity.this,SearchActivity.class));
                    break;
                case R.id.settings:
                    startActivity(new Intent(MainActivity.this,SettingsActivity.class));
                    break;
            }


            return false;
        });

    }


    void setupAdapter(){
        pagerAdapter = new PagerAdapter(getSupportFragmentManager(), this);
        binding.container.setAdapter(pagerAdapter);
        binding.tabs.setupWithViewPager(binding.container);
        binding.container.setCurrentItem(1,false);

    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/java/com/example/breezil/fave/ui/main/MainActivity.kt
package com.example.breezil.fave.ui.main

import android.content.Intent
import android.content.SharedPreferences
import android.content.pm.ActivityInfo
import android.databinding.DataBindingUtil
import android.preference.PreferenceManager
import android.support.v4.app.Fragment
import android.os.Bundle
import android.support.v7.widget.Toolbar

import com.example.breezil.fave.R
import com.example.breezil.fave.databinding.ActivityMainBinding
import com.example.breezil.fave.ui.explore.SearchActivity
import com.example.breezil.fave.ui.adapter.PagerAdapter
import com.example.breezil.fave.ui.preference.SettingsActivity
import com.example.breezil.fave.utils.helpers.BottomNavigationHelper

import javax.inject.Inject

import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.DaggerAppCompatActivity
import dagger.android.support.HasSupportFragmentInjector
import okhttp3.logging.HttpLoggingInterceptor
import timber.log.Timber

class MainActivity : DaggerAppCompatActivity(), HasSupportFragmentInjector {


    lateinit var binding: ActivityMainBinding
    private var sharedPreferences: SharedPreferences? = null
    private var themeMode: Boolean = false


    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    private lateinit var pagerAdapter: PagerAdapter


    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
        themeMode = sharedPreferences!!.getBoolean(getString(R.string.pref_theme_key), false)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT

        if (themeMode) {
            setTheme(R.style.DarkNoActionTheme)
        } else {
            setTheme(R.style.AppNoActionTheme)
        }
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)




        setupAdapter()
        setupBottomNavigation()
        setSupportActionBar(binding.mainToolbar as Toolbar)
        supportActionBar!!.title = getString(R.string.app_name)

        val logging = HttpLoggingInterceptor { message -> Timber.tag(getString(R.string.okhttp)).d(message) }
        logging.redactHeader(getString(R.string.authorization))
        logging.redactHeader(getString(R.string.cookie))
    }

    private fun setupBottomNavigation() {

        BottomNavigationHelper.disableShiftMode(binding.bottomNavViewBar)


        val menu = binding.bottomNavViewBar.menu
        val menuItem = menu.getItem(0)
        menuItem.isChecked = true


        /*
         * here sets the navigations to its corresponding activities
         */
        binding.bottomNavViewBar.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {

                R.id.main -> {
                }
                R.id.search -> {
                    startActivity(Intent(this@MainActivity, SearchActivity::class.java))
                    finish()
                }
                R.id.settings -> {
                    startActivity(Intent(this@MainActivity, SettingsActivity::class.java))
                    finish()
                }
            }


            false
        }

    }


    internal fun setupAdapter() {
        pagerAdapter = PagerAdapter(supportFragmentManager, this)
        binding.container.adapter = pagerAdapter
        binding.tabs.setupWithViewPager(binding.container)
        binding.container.setCurrentItem(1, false)

    }

    override fun supportFragmentInjector(): AndroidInjector<Fragment>? {
        return dispatchingAndroidInjector
    }
}
